server/tests/integration_test.rs

```diff
@@ -0,0 +1,765 @@
+
+    let parent_directory = current_executable
+        .parent()
+        .expect("Failed to get the parent directory.");
+
+    let binary_directory = parent_directory
+        .parent()
+        .expect("Failed to get the directory of the binary.");
+
+    binary_directory.to_owned()
+}
+
+/// Start the binary built for integration testing.
+fn start_binary(binary: &str) -> process::Command {
+    // Create path to binary
+    let mut path = get_binary_directory();
+    path.push(binary);
+    path.set_extension(env::consts::EXE_EXTENSION);
+
+    assert!(path.exists());
+
+    // Create command process
+    process::Command::new(path.into_os_string())
+}
+
+/// Start a new Arrow Flight Server to simulate a server for the integration tests.
+fn start_arrow_flight_server(dir: &Path) -> Child {
+    let mut process = start_binary("modelardbd")
+        .arg(dir)
+        .stdout(process::Stdio::piped())
+        .spawn()
+        .expect("Failed to start Arrow Flight Server");
+
+    let mut line = String::new();
+
+    let mut output = BufReader::new(process.stdout.as_mut().unwrap());
+
+    // Ensure that the process has fully started.
+    while !line.contains("Starting Apache Arrow Flight") {
+        output
+            .read_line(&mut line)
+            .expect("Could not read line of process.");
+    }
+
+    return process;
+}
+
+/// Create a new Arrow Flight client to access the endpoints in the ModelarDB server.
+fn create_flight_service_client(
+    rt: &Runtime,
+    host: &str,
+    port: u16,
+) -> Result<FlightServiceClient<Channel>, Box<dyn Error>> {
+    let address = format!("grpc://{}:{}", host, port);
+
+    rt.block_on(async {
+        let fsc = FlightServiceClient::connect(address).await?;
+        Ok(fsc)
+    })
+}
+
+/// Create a normal table or model table with/without tags in the
+/// ModelarDB server with Arrow Flight using the `do_action()` SQL Parser.
+/// mode options: NormalTable/ModelTable/ModelTableNoTag
+fn create_table(
+    rt: &Runtime,
+    client: &mut FlightServiceClient<Channel>,
+    table_name: &str,
+    mode: &str
+) {
+    let cmd = match mode{
+        "NormalTable" => format!(
+            "CREATE TABLE {}(timestamp TIMESTAMP, values REAL, metadata REAL)",
+            table_name
+        ),
+        "ModelTable" => format!(
+            "CREATE MODEL TABLE {}(timestamp TIMESTAMP, value FIELD, tag TAG)",
+            table_name
+        ),
+        "ModelTableNoTag" => format!(
+            "CREATE MODEL TABLE {}(timestamp TIMESTAMP, value FIELD)",
+            table_name
+        ),
+        _ => panic!("No mode selected in create_table.")
+    };
+
+    let action = Action {
+        r#type: "CommandStatementUpdate".to_string(),
+        body: cmd.into_bytes(),
+    };
+
+    rt.block_on(async {
+        client.do_action(Request::new(action)).await.expect("Could not create model table.");
+    })
+}
+
+/// Create the FlightDescriptor needed to initialize the stream.
```

Comment:
This comment does not seem to accurately describe what the function does.

Reply:
Fixed.